Sources/ArgumentParser/Parsable Properties/Argument.swift

```diff
@@ -83,6 +83,46 @@ extension Argument where Value: ExpressibleByArgument {
   }
 }
 
+/// The strategy to use when parsing multiple values from `@Option` arguments
+/// into an array.
+public enum ArgumentArrayParsingStrategy {
+  /// Parse only unprefixed values from the command-line input, ignoring
+  /// any inputs that have a dash prefix.
+  ///
+  /// For example, for a parsable type defined as following:
+  ///
+  ///     struct Options: ParsableArguments {
+  ///         @Flag() var verbose: Bool
+  ///         @Argument(parsing: .remaining) var words: [String]
+  ///     }
+  ///
+  /// Parsing the input `--verbose one two` or `one two --verbose` would result
+  /// in `Options(verbose: true, words: ["one", "two"])`. Parsing the input
+  /// `one two --other` would result in an unknown option error for `--other`.
+  ///
+  /// This is the default strategy for parsing argument arrays.
+  case remaining
+  
+  /// Parse all remaining inputs after parsing any known options or flags,
+  /// including dash-prefixed inputs.
+  ///
+  /// For example, for a parsable type defined as following:
+  ///
+  ///     struct Options: ParsableArguments {
+  ///         @Flag() var verbose: Bool
+  ///         @Argument(parsing: .remainingValues) var words: [String]
+  ///     }
+  ///
+  /// Parsing the input `--verbose one two --other` would include the `--other`
+  /// flag in `words`, resulting in
+  /// `Options(verbose: true, words: ["one", "two", "--other"])`.
+  ///
+  /// - Note: This parsing strategy can be surprising for users, particularly
+  ///   when combined with options and flags. Use care when specifying the
+  ///   `unconditionalRemaining` parsing strategy.
```

Comment:
I’d explicitly call out the alternative here:
```suggestion
  /// - Note: This parsing strategy can be surprising for users, particularly
  ///   when combined with options and flags. Use care when specifying the
  ///   `unconditionalRemaining` parsing strategy.
  ///   It’s generally recommended to use `.remaining`, and tell users to
  ///   _terminate_ options with `--` if any arguments may start with `-` or `--`,
  ///   i.e. use `.remaining` and tell users to call the tool with the input
  ///   `--verbose -- one two --other`.
```